pkg/graveler/delete_sensor.go

```diff
@@ -0,0 +1,154 @@
+package graveler
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/treeverse/lakefs/pkg/logging"
+)
+
+const (
+	callbackChannelSize = 1000
+	graceDuration       = 30 * time.Second
+)
+
+type DeleteSensorCB func(repositoryID RepositoryID, branchID BranchID, stagingTokenID StagingToken, inGrace bool)
+
+type StagingTokenCounter struct {
+	StagingTokenID StagingToken
+	Counter        int
+}
+
+type stagingTokenData struct {
+	repositoryID   RepositoryID
+	branchID       BranchID
+	stagingTokenID StagingToken
+}
+
+func (s *stagingTokenData) CombinedKey() string {
+	return fmt.Sprintf("%s:%s", s.repositoryID, s.branchID)
+}
+
+type DeleteSensor struct {
+	ctx       context.Context
```

Comment:
Every task should probably have its own context.  For instance, that allows logs to carry a request ID.

Reply:
Thanks! looks like leftovers, it isn't used